cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -69,6 +74,14 @@ func CreateSelfHostedControlPlane(cfg *kubeadmapi.MasterConfiguration, client *c
 		return err
 	}
 
+	if err := launchSelfHostedProxy(cfg, client); err != nil {
```

Comment:
What's the purpose of launching proxy here? How does it deploy kube-proxy before this PR?